libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialogBot.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Dialogs.Adaptive.Generators;
+using Microsoft.Bot.Builder.Dialogs.Declarative.Resources;
+using Microsoft.Bot.Builder.Skills;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive
+{
+    /// <summary>
+    /// An <see cref="IBot"/> implementation that hosts an <see cref="AdaptiveDialog"/>.
+    /// </summary>
+    public class AdaptiveDialogBot : IBot
```

Comment:
Yes those assumptions and you missed a pair of closely related assumptions that I think are significant:

- AdaptiveDialogs and ActivityHandler are mutually exclusive: if you are using AdaptiveDialogs you should **not** use ActivityHandler because there is no point (well technically, there is _almost_ no point)
- There is no coded AdaptiveDialogs, you always load AdaptiveDialogs from a ResourceExplorer

The assumptions around the various "views" on storage are the assumptions DialogManager makes (OK I extended this to include SkillConversationIdFactory but that is not much of a stretch in this context).

In practical terms, my thought was that this assumption comes down to an additional constructor. And I was thinking that was something we could add later if necessary.

I think we could pull the configuration dependency into the Integration package. ("integration.runtime" goes away right?) That extra layering would be consistent with what we have done in the past. But we did that only because of the platform issues and no-one is running Composer on WebApi so in that light this coupling seems reasonable. Beside which this WebApi thing comes down to IConfiguration in this case and when I redid the WebApi sample I added a couple of lines of code in the Controller that created the IConfiguration from the old style global mechanism.

Anyhow in practical terms,  this assumption comes down to an additional constructor. Same reasoning applies.

So I could:
- add more constructors
- and add those additional constructor via a derived class

incidentally why does the adaptive runtime Register the OnQnAMatch type?

Reply:
Of note on the assumption of UserState and Conversation state:

1) The current Runtime also makes this assumption.  It allows for the DI of IStorage, but always uses these classes.  On the assumption that we probably knows the best way to wire up a bot.
2) I somewhat feel, though I could be debated off this stance, DotNet DI is often used for DI's sake.  This could be another case of this.  Are UserState and ConversationState ever replaced with anything else?  Isn't it pretty safe to say that these provide the functionality we always want, so why make everyone wire it up?

Reply:
There are cases where the default keys for UserState and ConversationState do not suffice:
https://github.com/OfficeDev/BotBuilder-MicrosoftTeams-dotnet/blob/master/CSharp/Microsoft.Bot.Builder.Teams/StateStorage/TeamSpecificConversationState.cs#L41

Reply:
So in the last rev I made these pluggable - however to @tracyboehrer point we were hardcoding these in the runtime - so  should this logic be included in the runtime?

In which case why wouldn't we just include it in the regular ConversationState?

In other words that just seems like a bug in ConversationState.

Anyhow the argument for allowing these to be pluggable is that you might decide to use different underlying storage for user and conversation state. this argument is tenuous to say the least, however, lets just go with that for now.